compiler/rustc_errors/src/lib.rs

```diff
@@ -768,13 +768,7 @@ impl DiagCtxt {
                     format!("invalid level in `stash_diagnostic`: {:?}", diag.level),
                 );
             }
-            Error => {
-                // This `unchecked_error_guaranteed` is valid. It is where the
-                // `ErrorGuaranteed` for stashed errors originates. See
-                // `DiagCtxtInner::drop`.
-                #[allow(deprecated)]
-                Some(ErrorGuaranteed::unchecked_error_guaranteed())
-            }
+            Error => Some(self.span_delayed_bug(span, "stashing {key:?}")),
```

Comment:
@nnethercote I needed to change this so I could debug the origin of stashed errors at all.

Reply:
created a separate PR for this https://github.com/rust-lang/rust/pull/122194